### PR TITLE
Feature: password reset

### PR DESCRIPTION
**Pull request by shamoon**
## Proposed change

Seriously this is the last one from me for now. This implements a password reset page, really just using that which is built into Django and spiffing up the templates.  We could use the email server for other things down the road...

One thing I wasnt sure about, since it requires an smtp server I had the template hide the "Forgot your password?" link, its currently set to check `EMAIL_HOST` or `EMAIL_HOST_USER`, I figured there might actually be a situation where someones using a localhost for smtp.

Screenshots below.

Screenshot 2023-09-28 at 10 55 56 AM

Screenshot 2023-09-28 at 10 56 09 AM

Screenshot 2023-09-28 at 10 56 22 AM
Screenshot 2023-09-28 at 10 57 05 AM
Screenshot 2023-09-28 at 10 57 25 AM

Fixes #3362
Fixes #4269

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (please explain):

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [ ] If applicable, I have included testing coverage for new code in this PR, for [backend](https://docs.paperless-ngx.com/development/#testing) and / or [front-end](https://docs.paperless-ngx.com/development/#testing-and-code-style) changes.
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [x] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -1301,3 +1301,32 @@ started by the container.
...
+
+#### [`PAPERLESS_EMAIL_HOST=<str>`](#PAPERLESS_EMAIL_HOST) {#PAPERLESS_EMAIL_HOST}
+
+: Defaults to 'localhost'.
+
+#### [`PAPERLESS_EMAIL_PORT=<int>`](#PAPERLESS_EMAIL_PORT) {#PAPERLESS_EMAIL_PORT}
+
+: Defaults to port 25.
+
+#### [`PAPERLESS_EMAIL_USER=<str>`](#PAPERLESS_EMAIL_USER) {#PAPERLESS_EMAIL_USER}
```
Comment (stumpylog) [inline docs/configuration.md:1318]: This name doesn't match the `settings.py` environment name.

File: docs/configuration.md [hunk 2]
```diff
@@ -1301,3 +1301,32 @@ started by the container.
...
+
+: Defaults to ''.
+
+#### [`PAPERLESS_EMAIL_PASSWORD=<str>`](#PAPERLESS_EMAIL_PASSWORD) {#PAPERLESS_EMAIL_PASSWORD}
```
Comment (stumpylog) [inline docs/configuration.md:1322]: Nor this one

**Review by stumpylog — CHANGES_REQUESTED** [review]
A couple little things.  Didn't attempt to create a server to fully test.  Curious too if there can be any sort of check the link was sent correctly?  Or settings validation so the smtp server is known to be configured correctly?

File: docs/configuration.md [hunk 3]
```diff
@@ -1301,3 +1301,32 @@ started by the container.
     v1.9.2. A one-time migration is performed for users who have this
     setting set. This setting is always ignored if the corresponding
     frontend setting has been set.
+
+## Email sending
+
+Setting an SMTP server for the backend will allow you to reset your
+password. All of these options come from their similarly-named [Django settings](https://docs.djangoproject.com/en/1.8/ref/settings/#email-host)
```
[the remark this answers is not in the document]
Comment (shamoon) [reply docs/configuration.md:1308]: ha thanks

**Comment by shamoon** [comment]
Ah thanks, fixed those things (I swear I double-checked them, too much copypasta)

As for your qs:

- I dont know about the first part, again this comes straight from Django. I know if you try to send an email without a valid server you do get an [ugly] error in the browser.
- Validating the server I guess would mean actually connecting to it? Im not sure, is that something you would do in system checks? I dont really know how you would do it here? As you can see, we dont explicitly create these views, etc. we're just using templates for the built-in views, I think.

Totally open if you see how to do either, of course

**Comment by shamoon** [comment]
@stumpylog I added a system check for this, let me know what you think, no big deal to revert that part if youre not into it.

**Comment by stumpylog** [comment]
I think if there's a big exception of some sort in the logs, that is enough.  The thing with the checks, they run a lot.  So if a user wants to export, but their SMTP server is having problems, they couldn't as that check would fail.  And it's network calls, which could be pretty slow.

**Comment by shamoon** [comment]
The exception isnt in the logs, its in the browser, e.g.:

**Comment by shamoon** [comment]
Sorry, accidentally sent comment. The exception is in the logs and in the browser, e.g.:
Screenshot 2023-09-29 at 4 34 44 PM

I reverted the system check 👍

**Comment by stumpylog** [comment]
Curious, with `PAPERLESS_EMAIL_HOST_USER` defined, I get the link.  But no error, despite having no server configured that I know of.

Also (and I should have thought of this before), who's password does this reset?  Does it lookup via email the user?

**Comment by shamoon** [comment]
Sorry if Im misunderstanding. You only get the error after trying to submit the reset. If either of those are set it shows you the link, the idea being "this instance looks like it setup email" (but not validated).

And yea, uses email

**Comment by shamoon** [comment]
Edit: error is after (trying to) sending the email, not the reset pw

**Comment by stumpylog** [comment]
It must be from putting in an email which doesn't have a user mapped to it.  Django policy in that case is no error messages, based on the [note here](https://docs.djangoproject.com/en/4.2/topics/auth/default/#django.contrib.auth.views.PasswordResetView)

**Comment by shamoon** [comment]
Ah, yea now I get it. Yea thats a common thing on login forms I think, dont tell the attempter why it failed ("username not found", "wrong password", etc. all are just giving potential 'hints' to attackers)

**Review by stumpylog — APPROVED** [review]

**Pull request merged**